rest/data/project.go

```diff
@@ -389,3 +389,54 @@ func (pc *DBProjectConnector) GetProjectFromFile(ctx context.Context, pRef model
 	}
 	return model.GetProjectFromFile(ctx, opts)
 }
+
+func DeleteBranch(ctx context.Context, pRef *model.ProjectRef) error {
+	if pRef.IsHidden() {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
```

Comment:
Should we return the status code so that the graphql resolver can use this rather than wrapping it with internal server error?

Reply:
I ended up modifying the GraphQL resolver so it can capture the correct error — let me know if you want me to try something else though!